passes/builders.nim

```diff
@@ -48,6 +48,15 @@ func add*[T](bu: var Builder[T], nodes: openArray[TreeNode[T]]) =
     inc bu.buf[bu.parent].val
   bu.buf.add nodes
 
+func add*[T](bu: var Builder[T], trees: openArray[seq[TreeNode[T]]]) =
+  ## Appends all `trees` to the current sub-tree. The trees must each either
+  ## represent a single atomic node, or a complete subtree.
+  assert trees.len > 0
+  if bu.parent != -1:
+    inc bu.buf[bu.parent].val
+  for t in trees.items:
+    bu.buf.add t
```

Comment:
Not sure if we should add this overload, but it came up here and a bunch in the `If` branch. I think this is going to be a relatively common operation, but there is some minor merit to seeing the explicit `for ...`. This isn't a hard requirement for me.

Reply:
I think it makes sense to have, but given that it's mostly useful to `source2il`, I'd move it there. For ergonomic reasons, I'd also drop the assertion (so that passing empty statements lists to `add` works).

Reply:
Cool, I moved it over and dropped the assertion plus some other inner builder access.